packages/ketchup/src/components/kup-activity-timeline/kup-activity-timeline.tsx

```diff
@@ -0,0 +1,393 @@
+        return {
+            column: this.data.columns.find(
+                (column) => column.name === activityData.columnName
+            ),
+            row: this.data.rows.find((row) => row.id === activityData.cellId)
+                ?.cells?.[activityData.columnName],
+        };
+    }
+
+    onActivityClick(event: UIEvent, activityData: KupActivityTimelineData) {
+        this.kupClick.emit({
+            comp: this,
+            id: this.rootElement.id,
+            details: {
+                ...this.getActivity(activityData),
+                originalEvent: event,
+                type: KupActivityTimelineAction.onClick,
+            },
+        });
+    }
+
+    onActivityContextMenu(
+        event: UIEvent,
+        activityData: KupActivityTimelineData
+    ) {
+        this.kupContextMenu.emit({
+            comp: this,
+            id: this.rootElement.id,
+            details: {
+                ...this.getActivity(activityData),
+                originalEvent: event,
+                type: KupActivityTimelineAction.onRightClick,
+            },
+        });
+    }
+
+    activityItem(activity: KupActivity) {
+        return (
+            <div class="atm-row">
+                {activity.columns.map((column: KupActivityTimelineData) => (
+                    <div class="atm-col">
+                        <div class="atm-inner-col">
+                            <h3>{column.title}:</h3>
+                        </div>
+                        <div
+                            class="atm-inner-col"
+                            onClick={(e) => {
+                                e.stopPropagation();
+                                this.onActivityClick(e, column);
+                            }}
+                            onContextMenu={(e) => {
+                                e.stopPropagation();
+                                this.onActivityContextMenu(e, column);
+                            }}
+                        >
+                            <p>{column.value}</p>
+                        </div>
+                    </div>
+                ))}
+            </div>
+        );
+    }
+
+    calendarIcon(): VNode {
+        return (
+            <FImage
+                color="white"
+                resource={'calendar'}
+                wrapperClass="activity-timeline-item__icon"
+                sizeX="20px"
+                sizeY="32px"
+            />
+        );
+    }
+    /*-------------------------------------------------*/
+    /*          L i f e c y c l e   H o o k s          */
+    /*-------------------------------------------------*/
+
+    componentWillLoad() {
+        this.#kupManager.debug.logLoad(this, false);
+        this.#kupManager.theme.register(this);
+    }
+
+    componentDidLoad() {
+        this.kupReady.emit({
+            comp: this,
+            id: this.rootElement.id,
+        });
+        this.#kupManager.debug.logLoad(this, true);
```

Comment:
The emission of the `ready` event and the `logLoad` hook should be called at the bottom of the `componentDidLoad` hook, to accurately signal the readiness.